File: 008_Array_Part_1/sizeOf.py
import sys

# 1. Static list
arr = [10, 20, 30, 40, 50]

# Number of elements
print("Number of elements in list:", len(arr))

# Memory size of the list object
print("Memory size of list object (bytes):", sys.getsizeof(arr))

# Memory size of each element
print("Size of each element (bytes):", sys.getsizeof(arr[0]))

# Total memory of all elements (approx)
total_memory = sys.getsizeof(arr) + sum(sys.getsizeof(i) for i in arr)
print("Total memory used by list and its elements:", total_memory)

# Output
print("List elements:", *arr)

# 2. Dynamic input
n = int(input("\nEnter size of dynamic list: "))
dyn_arr = []

print(f"Enter {n} elements:")
for _ in range(n):
    dyn_arr.append(int(input()))

print("Dynamic list elements:", *dyn_arr)
print("Length of dynamic list:", len(dyn_arr))
print("sys.getsizeof(dyn_arr):", sys.getsizeof(dyn_arr))

File: 008_Array_Part_1/sizeOf.cpp
#include<iostream>
using namespace std;

int main() {
    // 1. Static array
    int arr[] = {10, 20, 30, 40, 50};

    // Total size in bytes
    cout << "Total size of array in bytes: " << sizeof(arr) << endl;

    // Size of one element
    cout << "Size of one element: " << sizeof(arr[0]) << endl;

    // Number of elements
    int size = sizeof(arr) / sizeof(arr[0]);
    cout << "Number of elements in array: " << size << endl;

    // Output using loop
    cout << "Array elements: ";
    for (int i = 0; i < size; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;

    // 2. Dynamic array (using pointers)
    int n;
    cout << "\nEnter size for dynamic array: ";
    cin >> n;

    int* dynArr = new int[n];
    cout << "Enter " << n << " elements: ";
    for (int i = 0; i < n; i++) {
        cin >> dynArr[i];
    }

    cout << "Dynamic array elements: ";
    for (int i = 0; i < n; i++) {
        cout << dynArr[i] << " ";
    }
    cout << "\nNumber of elements: " << n << endl;

    // sizeof won't work correctly for dynamic array
    cout << "sizeof(dynArr): " << sizeof(dynArr) << " (pointer size, not array size)\n";

    delete[] dynArr;
    return 0;
}